core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java

```diff
@@ -63,15 +51,12 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
 
         String value = null;
         float boost = 1.0f;
-        Fuzziness fuzziness = DEFAULT_FUZZINESS;
-        int prefixLength = FuzzyQuery.defaultPrefixLength;
-        int maxExpansions = FuzzyQuery.defaultMaxExpansions;
-        boolean transpositions = false;
+        Fuzziness fuzziness = FuzzyQueryBuilder.DEFAULT_FUZZINESS;
+        int prefixLength = FuzzyQueryBuilder.DEFAULT_PREFIX_LENGTH;
+        int maxExpansions = FuzzyQueryBuilder.DEFAULT_MAX_EXPANSIONS;
+        boolean transpositions = FuzzyQueryBuilder.DEFAULT_TRANSPOSITIONS;
+        String rewrite = null;
         String queryName = null;
-        MultiTermQuery.RewriteMethod rewriteMethod = null;
-        if (parseContext.isFilter()) {
-            rewriteMethod = MultiTermQuery.CONSTANT_SCORE_REWRITE;
-        }
```

Comment:
Thanks, seems that unfortunately we need to re-set it then when we're in filter context. @alexksikes I guess that means we have to go back to using the String constants from `QueryParser#parseRewriteMethod`. I was wondering if it makes sense to factor out the constants used there into a separate enum or a fixed list of constants to make the dependency of the Parser to those constants more explicit. I see that the second part of `QueryParser#parseRewriteMethod` also does some suffix-magix with the values passed in, but at least for the constant we use in FuzzyQueryParser it would be good to link it somehow to the code there. Or whats your take on this?